core/src/main/java/dev/streamx/cli/ingestion/BasePublishUnpublishCommand.java

```diff
@@ -0,0 +1,74 @@
+package dev.streamx.cli.ingestion;
+
+import dev.streamx.cli.exception.IngestionClientException;
+import dev.streamx.clients.ingestion.StreamxClient;
+import dev.streamx.clients.ingestion.exceptions.StreamxClientException;
+import jakarta.inject.Inject;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import picocli.CommandLine.ArgGroup;
+import picocli.CommandLine.Model.CommandSpec;
+import picocli.CommandLine.Spec;
+
+public abstract class BasePublishUnpublishCommand implements Runnable {
+
+  @ArgGroup(exclusive = false)
+  protected IngestionArguments ingestionArguments;
+
+  @Spec
+  protected CommandSpec spec;
+
+  @Inject
+  protected SchemaProvider schemaProvider;
+
+  @Inject
+  protected StreamxClientProvider streamxClientProvider;
+
+  protected abstract void performCommand(StreamxClient client) throws StreamxClientException;
+
+  @Override
+  public void run() {
+    StreamxClient client = createClient();
+
+    try {
+      performCommand(client);
+    } catch (StreamxClientException e) {
+      if (is404Error(e)) {
+        createExtendedIngestionClientException(e);
+      } else {
+        createIngestionClientException(e);
+      }
+    } finally {
+      try {
+        client.close();
+      } catch (StreamxClientException e) {
+        throw new IngestionClientException(e);
+      }
+    }
+  }
```

Comment:
This method is modified.
Now no validation beforehand.
The change is:
 - if publish/unpublish worked, then we return
 - if not - we collect available channel names and add them to exception message

Note: we cannot use try-with-resources as before, because:
 - if publish/unpublish throws StreamxClientException - the underlying httpClient gets closed
 - so we need to the call to get schemas while it is still open